worlds/wl4/regions.py

```diff
@@ -0,0 +1,315 @@
+    create_main_regions(world, location_table)
+    create_level_regions(world, location_table)
+
+
+def create_main_regions(world: WL4World, location_table: Set[str]):
+    basic_region = basic_region_creator(world, location_table)
+
+    def passage_region(passage: Passage):
+        return basic_region(passage.long_name())
+
+    def minigame_shop(passage: Passage):
+        return basic_region(f'{passage.short_name()} Minigame Shop')
+
+    def boss_region(passage: Passage, boss_name: str):
+        return basic_region(f'{passage.long_name()} Boss',
+                            [boss_name, *(f'{boss_name} - 0:{time}' for time in ('15', '35', '55'))])
+
+    menu_region = basic_region('Menu')
+
+    passage_regions = (passage_region(passage) for passage in Passage)
+    minigame_shops = (minigame_shop(passage) for passage in itertools.islice(Passage, 1, None))
+
+    # spoiled_rotten = boss_region(Passage.ENTRY, 'Spoiled Rotten')
+    cractus = boss_region(Passage.EMERALD, 'Cractus')
+    cuckoo_condor = boss_region(Passage.RUBY, 'Cuckoo Condor')
+    aerodent = boss_region(Passage.TOPAZ, 'Aerodent')
+    catbat = boss_region(Passage.SAPPHIRE, 'Catbat')
+    golden_diva = boss_region(Passage.GOLDEN, 'Golden Diva')
+
+    sound_room = basic_region('Sound Room', ['Sound Room - Emergency Exit'])
+
+    world.multiworld.regions += [
+        menu_region,
+        *passage_regions,
+        *minigame_shops,
+        # spoiled_rotten,
+        cractus,
+        cuckoo_condor,
+        aerodent,
+        catbat,
+        golden_diva,
+        sound_room,
+    ]
+
+
+# Eventually I'll want to make trees out of the regions in the levels for
+# diamond shuffle, but sequences will work for now
+regions_in_levels = {
+    'Hall of Hieroglyphs':   None,
+
+    'Palm Tree Paradise':    None,
+    'Wildflower Fields':     ['Before Sunflower', 'After Sunflower'],
+    'Mystic Lake':           ['Shore', 'Shallows', 'Depths'],
+    'Monsoon Jungle':        ['Upper', 'Lower'],
+
+    'The Curious Factory':   None,
+    'The Toxic Landfill':    None,
+    '40 Below Fridge':       None,
+    'Pinball Zone':          ['Early Rooms', 'Jungle Room', 'Late Rooms', 'Escape'],
+
+    'Toy Block Tower':       None,
+    'The Big Board':         None,
+    'Doodle Woods':          None,
+    'Domino Row':            ['Before Lake', 'After Lake'],
+
+    'Crescent Moon Village': ['Upper', 'Lower'],
+    'Arabian Night':         ['Town', 'Sewer'],
+    'Fiery Cavern':          ['Flaming', 'Frozen'],
+    'Hotel Horror':          ['Hotel', 'Switch Room'],
+
+    'Golden Passage':        ['Passage', 'Keyzer Area'],
+}
+
+
+def get_region_names(level_name: str, merge: bool = False) -> Sequence[str]:
+    entrance = f'{level_name} (entrance)'
+    regions = regions_in_levels[level_name]
+    if regions and not merge:
+        regions = (f'{level_name} - {region}' for region in regions)
+    else:
+        regions = (level_name,)
```

Comment:
You use lists everywhere in  `regions_in_levels`, so you should do the same here
```suggestion
        regions = [level_name]
```